### PR TITLE
feat: Added ColorPicker component (to ColorField)

### PR DESCRIPTION
**Pull request by Dreamdealer**
### This PR:

**Backwards compatible additions** ✨
- Added new ColorPicker component to Bricks which implements react-color (custom version)
- Added ColorPicker to ColorField component
- Made the transparent swatch in the ColorPicker optional

**Checklist** 🛡
- [x] I have exported my addition from `src/index.ts` (check if not applicable).
- [x] Appropriate tests have been added for my functionality (check if not applicable).
- [ ] A designer has seen and approved my changes (tag `@LuukHorsmans` or `@RianneSchaekens` for a design review when applicable).
- [x] I have tested my addition in all supported browsers and for responsiveness (Chrome, Firefox, Safari, Edge, IE11 and mobile browsers).
- [x] Appropriate documentation has been written (check if not applicable).

<details>
  <summary>ℹ️ Conventional commit help</summary>

  - Use `fix: ...` for patches
  - Use `feat: ...` for a minor change
  - Use `feat!: ...` for features with breaking changes
  - Optionally you can use a scope `feat(SomeContainer)!: ...`
  - A breaking change **must** have `BREAKING CHANGE: ...` in the commit message
  - You can use multiple `BREAKING CHANGE: ...` lines in the commit message
  - Use `build`, `chore`, `ci`, `docs`, `perf`, `refactor`, `style`, or `test`  for other types

  More info see full [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/).

  _Tip: make sure the first commit message is the one you want to merge, because that's the one Github picks for squashing._
</details>

### CODE REVIEW

File: packages/components/src/components/ColorPicker/index.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    margin-top: -6px;
+    margin-left: -6px;
+`;
+const HiddenSaturationPointer = styled.div`
+    width: 0;
+    height: 0;
+`;
+
+const ColorPicker: FC<PropsType> = props => {
+    const { children, ...propsWithoutChildren } = { ...props };
```
Comment (CarloPalinckx) [inline packages/components/src/components/ColorPicker/index.tsx:47]: `const { children, ...propsWithoutChildren } = props;` should work as well

File: packages/components/src/components/ColorPicker/index.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            placement={'bottom-start'}
+            arrowStyle={{
+                left: '5px',
+            }}
+            onClickOutside={() => {
+                props.onClickOutside();
+            }}
+            renderContent={() => (
+                <Box direction="column" padding={[12]} width="230px">
+                    <Box height="130px" width="100%" style={{ position: 'relative' }}>
```
Comment (CarloPalinckx) [inline packages/components/src/components/ColorPicker/index.tsx:62]: position is a prop on Box

File: packages/components/src/components/ColorPicker/index.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                                        props.onChange({
+                                            hex: 'transparent',
+                                            hsl: { a: 0, h: 0, l: 0, s: 0 },
+                                            rgb: { a: 0, r: 0, g: 0, b: 0 },
+                                        });
+                                    }}
+                                />
+                            </Box>
+                        )}
+                        <Box height="12px" width="100%" style={{ position: 'relative' }}>
```
Comment (CarloPalinckx) [inline packages/components/src/components/ColorPicker/index.tsx:84]: Same applies here

File: packages/components/src/components/ColorField/index.tsx [hunk 4]
```diff
@@ -42,60 +44,79 @@ const ColorField: FC<PropsType> = props => {
...
+                data-testid={`${props['data-testid']}-${TestIds.container}`}
+            >
+                <Box padding={[6, 12, 0, 0]}>
+                    <ColorDrop
+                        color={props.emptyIsTransparent && isTransparent(props.value) ? 'transparent' : props.value}
+                        data-testid={`${props['data-testid']}-${TestIds.colorDrop}`}
+                        onClick={() => {
+                            setShow(!show);
+                        }}
+                        style={{ cursor: 'pointer' }}
```
Comment (CarloPalinckx) [inline packages/components/src/components/ColorField/index.tsx:70]: Maybe you could add this to the styled component of the color drop as a condition `if props.onClick -> cursor: pointer`

**Review by CarloPalinckx — CHANGES_REQUESTED** [review]
Few small nits, looks nice otherwise!

File: packages/components/src/components/ColorField/index.tsx (hunk 4 above)
Comment (Dreamdealer) [reply packages/components/src/components/ColorField/index.tsx:70]: Nice one!

**Review by CarloPalinckx — APPROVED** [review]

**Pull request merged**